CRM/Upgrade/Incremental/php/FiveFiftySeven.php

```diff
@@ -21,11 +21,24 @@
  */
 class CRM_Upgrade_Incremental_php_FiveFiftySeven extends CRM_Upgrade_Incremental_Base {
 
+  /**
+   * How many activities before the queries used here are slow. Guessing.
+   */
+  const ACTIVITY_THRESHOLD = 1000000;
+
   public function setPreUpgradeMessage(&$preUpgradeMessage, $rev, $currentVer = NULL) {
     if ($rev === '5.57.alpha1') {
-      if (CRM_Core_DAO::singleValueQuery('SELECT COUNT(id) FROM civicrm_activity WHERE is_current_revision = 0')) {
+      // The query on is_current_revision is slow if there's a lot of activities. So limit when it gets run.
+      $activityCount = CRM_Core_DAO::singleValueQuery('SELECT COUNT(id) FROM civicrm_activity');
```

Comment:
This is an improvement but actually `SELECT COUNT` is a slow query on large DBS. Despite the limitations `SELECT MAX(id) ` is better

Reply:
actually - sorry this is not the query I thought it was - I was thinking of those slow COUNT queries @totten added which decide whether to create snapshots

Reply:
Nope I was right the first time - this query WOULD be better with the `SELECT MAX(id) ` to speed it up